genie-web/src/main/java/com/netflix/genie/web/services/impl/JobResolverServiceImpl.java

```diff
@@ -672,4 +724,214 @@ private int resolveJobMemory(final JobRequest jobRequest, final Command command)
             .getRequestedJobMemory()
             .orElse(command.getMemory().orElse(this.defaultMemory));
     }
+
+    private boolean useV4ResourceSelection() {
+        // TODO: Flesh this out with dynamic behavior to aid migration between V3 and V4
+        return false;
+    }
+
+    private Command resolveCommand(final JobRequest jobRequest) throws GenieJobResolutionException {
+        final long start = System.nanoTime();
+        final Set<Tag> tags = Sets.newHashSet();
+        try {
+            final Criterion criterion = jobRequest.getCriteria().getCommandCriterion();
+            final Set<Command> commands = this.commandPersistenceService.findCommandsMatchingCriterion(criterion);
+            final Command command;
+            if (commands.isEmpty()) {
+                throw new GenieJobResolutionException("No command matching command criterion found");
```

Comment:
Error message here and on 746 are kinda the same (i like the latter more), maybe use exactly the same for sake of error classification systems?
Or make the 2 more different/precise.